### PR TITLE
Migrated List, Map, MultiMap, ReplicatedMap, Queue and Set tests to async/await

### PR DESCRIPTION
**Pull request by sudiptog81**
Migrated tests in `test/{List,Map,MultiMap,ReplicatedMap,Queue,Set}` to use the `async/await` syntax.

Related to #569.

### CODE REVIEW

**Comment by mdumandag** [comment]
verify

File: test/replicatedmap/ReplicatedMapProxyTest.js [hunk 1]
```diff
@@ -29,245 +29,168 @@ describe('ReplicatedMapProxyTest', function () {
...
-            })
-            .then(function () {
-                return rm.get('1000ttl');
-            })
-            .then(function (val) {
-                expect(val).to.be.null;
-            });
+    it('puts entry with 1000 ttl', async function () {
+        await rm.put('1000ttl', 'value', 1000);
+        await new Promise(function (resolve) {
```
Comment (mdumandag) [inline test/replicatedmap/ReplicatedMapProxyTest.js:67]: Minor: I know the test is like that before but we can use `promiseWaitMilliseconds` from the `test/Util.js`

**Review by mdumandag — CHANGES_REQUESTED** [review]
Thanks for the great effort @sudiptog81! The changes look pretty good. I added some comments, could you check and fix them?

**Comment by sudiptog81** [comment]
> Thanks for the great effort @sudiptog81! The changes look pretty good. I added some comments, could you check and fix them?

Sure. Will address the changes in the next commit.

File: test/map/MapProxyTest.js [hunk 2]
```diff
@@ -232,233 +204,176 @@ describe('MapProxyTest', function () {
...
+                const isLocked = await map.isLocked('key0');
+                expect(isLocked).to.be.false;
             });
 
-            it('keySet', function () {
-                return map.keySet().then(function (keySet) {
-                    return expect(keySet).to.deep.have.members([
+            it('keySet', async function () {
+                const keySet = await map.keySet();
+                expect(keySet).to.deep.have.members([
```
[the remark this answers is not in the document]
Comment (sudiptog81) [reply test/map/MapProxyTest.js:289]: Addressed.

File: test/map/MapProxyTest.js [hunk 3]
```diff
@@ -507,74 +421,50 @@ describe('MapProxyTest', function () {
...
-            it('tryRemove fail', function () {
-                return RC.executeOnController(cluster.id, generateLockScript(map.getName(), '"key0"'), 1)
-                    .then(function () {
-                        return map.tryRemove('key0', 200);
-                    })
-                    .then(function (success) {
-                        return expect(success).to.be.false;
-                    });
+            it('tryRemove fail', async function () {
+                return RC.executeOnController(cluster.id, generateLockScript(map.getName(), '"key0"'), 1);
```
[the remark this answers is not in the document]
Comment (sudiptog81) [reply test/map/MapProxyTest.js:465]: Addressed.

File: test/map/NearCachedMapTest.js [hunk 4]
```diff
@@ -39,195 +39,152 @@ describe('NearCachedMapTest', function () {
...
-                    .then(() => client2.shutdown())
-                    .then(() => RC.terminateCluster(cluster.id));
+            after(async function () {
+                await client1.shutdown();
+                await client2.shutdown();
+                return RC.terminateCluster(cluster.id);
             });
 
-            function getNearCacheStats(map) {
+            async function getNearCacheStats(map) {
```
[the remark this answers is not in the document]
Comment (sudiptog81) [reply test/map/NearCachedMapTest.js:66]: Addressed.

File: test/map/NearCachedMapTest.js [hunk 5]
```diff
@@ -39,195 +39,152 @@ describe('NearCachedMapTest', function () {
...
                 return map.nearCache.getStatistics();
             }
 
-            function expectStats(map, hit, miss, entryCount) {
-                const stats = getNearCacheStats(map);
+            async function expectStats(map, hit, miss, entryCount) {
```
[the remark this answers is not in the document]
Comment (sudiptog81) [reply test/map/NearCachedMapTest.js:70]: Addressed.

File: test/map/NearCachedMapTest.js [hunk 6]
```diff
@@ -39,195 +39,152 @@ describe('NearCachedMapTest', function () {
...
-                });
-            });
-
-            it('get returns null if the entry was removed by another client', function () {
+            it('second get should hit', async function () {
+                await map1.get('key0');
+                const val = await map1.get('key0');
+                const stats = await getNearCacheStats(map1);
+                expect(val).to.equal('val0');
+                expect(stats.missCount).to.equal(1);
```
[the remark this answers is not in the document]
Comment (sudiptog81) [reply test/map/NearCachedMapTest.js:82]: Addressed.

File: test/map/NearCachedMapTest.js [hunk 7]
```diff
@@ -39,195 +39,152 @@ describe('NearCachedMapTest', function () {
...
+                expect(stats.hitCount).to.equal(1);
+            });
+
+            it('remove operation removes entry from near cache', async function () {
+                await map1.get('key1');
+                await map1.remove('key1');
+                const val = await map1.get('key1');
+                const stats = await getNearCacheStats(map1);
+                expect(val).to.be.null;
+                expect(stats.hitCount).to.equal(0);
```
[the remark this answers is not in the document]
Comment (sudiptog81) [reply test/map/NearCachedMapTest.js:93]: Addressed.

File: test/map/NearCachedMapTest.js [hunk 8]
```diff
@@ -39,195 +39,152 @@ describe('NearCachedMapTest', function () {
...
+                expect(stats.entryCount).to.equal(1);
+            });
+
+            it('update invalidates the near cache', async function () {
+                await map1.get('key1');
+                await map1.put('key1', 'something else');
+                const val = await map1.get('key1');
+                const stats = await getNearCacheStats(map1);
+                expect(val).to.be.equal('something else');
+                expect(stats.hitCount).to.equal(0);
```
[the remark this answers is not in the document]
Comment (sudiptog81) [reply test/map/NearCachedMapTest.js:104]: Addressed.

File: test/queue/QueueProxyTest.js [hunk 9]
```diff
@@ -67,161 +58,132 @@ describe('QueueProxyTest', function () {
...
+        expect(s).to.equal(11);
     });
 
-    it('add throws if queue is full', function () {
-        return offerToQueue(5, 'new').then(function () {
-            return expect(queue.add('excess_item')).to.eventually.rejected;
-        });
+    it('add throws if queue is full', async function () {
+        await offerToQueue(5, 'new');
+        expect(queue.add('excess_item')).to.eventually.rejected;
```
[the remark this answers is not in the document]
Comment (sudiptog81) [reply test/queue/QueueProxyTest.js:89]: Addressed.

File: test/replicatedmap/ReplicatedMapProxyTest.js (hunk 1 above)
Comment (sudiptog81) [reply test/replicatedmap/ReplicatedMapProxyTest.js:67]: Addressed.

File: test/map/MapProxyTest.js [hunk 10]
```diff
@@ -507,74 +421,50 @@ describe('MapProxyTest', function () {
...
-                return RC.executeOnController(cluster.id, generateLockScript(map.getName(), '"key0"'), 1)
-                    .then(function () {
-                        const promise = map.tryLock('key0', 1000);
-                        RC.executeOnController(cluster.id, generateUnlockScript(map.getName(), '"key0"'), 1);
-                        return promise;
-                    })
-                    .then(function (success) {
-                        return expect(success).to.be.true;
-                    });
+            it('tryLock_success with timeout', async function () {
```
[the remark this answers is not in the document]
Comment (sudiptog81) [reply test/map/MapProxyTest.js:435]: Addressed.

**Review by mdumandag — APPROVED** [review]
Looks good to me after the changes. Let me re-run the test suite and then we can merge

**Comment by mdumandag** [comment]
verify

File: test/map/NearCachedMapTest.js [hunk 11]
```diff
@@ -77,33 +77,24 @@ describe('NearCachedMapTest', function () {
             it('second get should hit', async function () {
                 await map1.get('key0');
                 const val = await map1.get('key0');
-                const stats = await getNearCacheStats(map1);
                 expect(val).to.equal('val0');
-                expect(stats.missCount).to.equal(1);
-                expect(stats.entryCount).to.equal(1);
-                expect(stats.hitCount).to.equal(1);
+                expectStats(map, 1, 1, 1);
```
Comment (mdumandag) [inline test/map/NearCachedMapTest.js:81]: The tests failed due to this line. This should be `map1` instead of `map`. Can you change that?

**Comment by mdumandag** [comment]
verify

**Comment by mdumandag** [comment]
Thanks a lot @sudiptog81 for the contribution. I am merging the PR :+1:

**Pull request merged**